extensions/amp-iframely/0.1/amp-iframely.js

```diff
@@ -0,0 +1,316 @@
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    /**
+     * Address of Iframely media (3rd party rich media or a summary card for the URL).
+     * @private {string}
+     */
+    this.src_ = null;
+
+    /**
+     * The main iFrame element that loads Iframely media.
+     * @private {?Element}
+     */
+    this.iframe_ = null;
+
+    /**
+     * ID of Iframely content, if available
+     * @private {?string}
+     */
+    this.widgetId_ = null;
+
+    /**
+     * Domain of Iframely CDN
+     * @private {?string}
+     */
+    this.base_ = null;
+
+    /**
+     * Alternatively, identify with Iframely CDN via `url` and `key` hash params
+     * @private {?string}
+     */
+    this.url_ = null;
+    /** @private {?string} */
+    this.key_ = null;
+
+    /**
+     * other data- options that will be passed into iFrame src
+     * @private {?Object}
+     * */
+    this.options_ = null;
+  }
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    /** Populating parameters once */
+    this.widgetId_ = this.element.getAttribute('data-id');
+    this.url_ = this.element.getAttribute('data-url');
+    this.key_ = this.element.getAttribute('data-key');
+    this.options_ = this.parseOptions_();
+    let domain = 'cdn.iframe.ly';
+    const requestedDomain = this.element.getAttribute('data-domain');
+    if (requestedDomain && this.isValidDomain_(requestedDomain)) {
+      domain = requestedDomain;
+    }
+    this.base_ = `https://${domain}/`;
+    this.parseAttributes_();
+  }
+
+  /**
+   * @param {boolean=} opt_onLayout
+   * @override
+   */
+  preconnectCallback(opt_onLayout) {
+    if (this.preconnect && this.preconnect.url_) {
+      this.preconnect.url(this.base_, opt_onLayout);
+    }
+  }
+
+  /** @override */
+  createPlaceholderCallback() {
+    /**
+     * Building placeholder for responsive layout without resizing
+     * or when requested via `data-img` attribute
+     */
+    const layout = this.getLayout();
+    if (
+      this.element.hasAttribute('data-img') ||
+      (layout === 'responsive' && !this.element.hasAttribute('resizable'))
+    ) {
+      /** using Iframely placeholder image */
+      const src = addParamsToUrl(
+        this.constructSrc_('/thumbnail'),
+        this.options_
+      );
+      return createElementWithAttributes(
+        this.element.ownerDocument,
+        'amp-img',
+        {
+          'src': src,
+          'placeholder': '',
+          'layout': 'fill',
+        }
+      );
+    }
+    return null;
+  }
+
+  /** @override */
+  layoutCallback() {
+    /** attach iFrame */
+    this.iframe_ = this.element.ownerDocument.createElement('iframe');
+    setStyle(this.iframe_, 'border', '0px');
+    this.iframe_.setAttribute(
+      'allow',
+      'encrypted-media *; accelerometer *; gyroscope *; picture-in-picture *; camera *; microphone *;'
```

Comment:
Could you please provide a justification for this attribute? List **each** of these, with its accompanying feature.

Reply:
We do not need allow attribute for Iframely's own features. The list represents what Iframely publishers required at the time of implementing this plugin 18 months ago. 

* `encrypted-media *; accelerometer *; gyroscope *; picture-in-picture *;` - come, for example, from YouTube
* `camera *; microphone *;` - required by VideoAsk.

Reply:
> `encrypted-media *; accelerometer *; gyroscope *; picture-in-picture *;` - come, for example, from YouTube

These aren't required on `amp-youtube` and the embed works fine. Could you explain further?

> `camera *; microphone *;` - required by VideoAsk.

Could you explain how?

Reply:
As embeds aggregator, we do not go into tech details of publishers implementations. We know that the embed codes that YouTube provides come with this allow attribute. Just go to any vid and copy-paste their HTML code. The same is for VideoAsk - they let record video replies in their iFrames.

Reply:
It's important for the top-level AMP document to give iframes the least possible power. It's not a matter of being technical implementation details, but user features.

There is no other AMP component that requires this allowlist and as such it cannot be submitted. You may search each property individually and you'll see that you can't find e.g. `encrypted-media *`.

Please remove these, and confirm that your embeds work fine.

Reply:
The publishers should have fallbacks in place if the permissions aren't granted. So we'll simply take it as the limitation of AMP and remove the `allow` attribute.

Reply:
Please update this change with:

- `allow`
- `measureIntesection`

And this PR should be ready to go.

Reply:
@alanorozco this should now be resolved. Thanks!